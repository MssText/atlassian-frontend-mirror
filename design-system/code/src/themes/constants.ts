export const CODE_FONT_SIZE = 12;
export const CODE_LINE_HEIGHT = '20px';

// css vars used to decrease font family being repeatedly declared inline
export const VAR_CODE_FONT_FAMILY = '--code-font-family';
export const VAR_CODE_FONT_FAMILY_ITALIC = `${VAR_CODE_FONT_FAMILY}-italic`;
export const VAR_CODE_LINE_NUMBER_BG_COLOR = '--line-number-bg-color';
