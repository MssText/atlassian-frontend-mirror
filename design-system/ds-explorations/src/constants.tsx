export const SPACING_SCALE = {
  'sp-25': 2,
  'sp-50': 4,
  'sp-75': 6,
  'sp-100': 8,
  'sp-200': 16,
  'sp-300': 24,
  'sp-400': 32,
  'sp-600': 48,
  'sp-800': 64,
};

export type GlobalSpacingToken = keyof typeof SPACING_SCALE;
