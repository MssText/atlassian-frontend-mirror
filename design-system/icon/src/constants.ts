import type { Size } from './types';

export const sizes: Record<Size, string> = {
  small: '16px',
  medium: '24px',
  large: '32px',
  xlarge: '48px',
};

export const sizeMap: Record<Size, Size> = {
  small: 'small',
  medium: 'medium',
  large: 'large',
  xlarge: 'xlarge',
};
