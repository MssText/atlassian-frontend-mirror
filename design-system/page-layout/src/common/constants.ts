// dimension vars
export const LEFT_PANEL_WIDTH = 'leftPanelWidth';
export const LEFT_SIDEBAR_WIDTH = 'leftSidebarWidth';
export const RIGHT_SIDEBAR_WIDTH = 'rightSidebarWidth';
export const RIGHT_PANEL_WIDTH = 'rightPanelWidth';
export const TOP_NAVIGATION_HEIGHT = 'topNavigationHeight';
export const BANNER_HEIGHT = 'bannerHeight';
export const LEFT_SIDEBAR_FLYOUT = 'leftSidebarFlyoutWidth';
export const DIMENSIONS = [
  LEFT_PANEL_WIDTH,
  RIGHT_PANEL_WIDTH,
  BANNER_HEIGHT,
  TOP_NAVIGATION_HEIGHT,
  LEFT_SIDEBAR_WIDTH,
  RIGHT_SIDEBAR_WIDTH,
];

// Grid area names
export const LEFT_PANEL = 'left-panel';
export const RIGHT_PANEL = 'right-panel';
export const BANNER = 'banner';
export const TOP_NAVIGATION = 'top-navigation';
export const CONTENT = 'content';
export const MAIN = 'main';
export const LEFT_SIDEBAR = 'left-sidebar';
export const RIGHT_SIDEBAR = 'right-sidebar';

// Default slot dimension values
export const DEFAULT_BANNER_HEIGHT = 56;
export const DEFAULT_TOP_NAVIGATION_HEIGHT = 56;
export const DEFAULT_LEFT_SIDEBAR_WIDTH = 240;
export const DEFAULT_RIGHT_SIDEBAR_WIDTH = 280;
export const DEFAULT_RIGHT_PANEL_WIDTH = 368;
export const DEFAULT_LEFT_PANEL_WIDTH = 368;

// Other constants
export const COLLAPSED_LEFT_SIDEBAR_WIDTH = 20;
export const MIN_LEFT_SIDEBAR_WIDTH = 80;
export const LEFT_SIDEBAR_FLYOUT_WIDTH = 240;
export const MIN_LEFT_SIDEBAR_DRAG_THRESHOLD = 200;
export const TRANSITION_DURATION = 300;
export const FLYOUT_DELAY = 200;
export const LEFT_SIDEBAR_EXPANDED_WIDTH = 'expandedLeftSidebarWidth';
export const PAGE_LAYOUT_LS_KEY = 'DS_PAGE_LAYOUT_UI_STATE';

// Data attributes
export const IS_SIDEBAR_DRAGGING = 'data-is-sidebar-dragging';
export const IS_SIDEBAR_COLLAPSING = 'data-is-sidebar-collapsing';
export const IS_FLYOUT_OPEN = 'data-is-flyout-open';
export const GRAB_AREA_LINE_SELECTOR = 'data-grab-area-line';
export const GRAB_AREA_SELECTOR = 'data-grab-area';
export const RESIZE_BUTTON_SELECTOR = 'data-resize-button';
export const RESIZE_CONTROL_SELECTOR = 'data-resize-control';
export const MAIN_SELECTOR = 'data-main';
export const LEFT_SIDEBAR_SELECTOR = 'data-left-sidebar';

export const DEFAULT_I18N_PROPS_SKIP_LINKS = 'Skip to:';

export const PAGE_LAYOUT_CONTAINER_SELECTOR = 'data-layout-container';
