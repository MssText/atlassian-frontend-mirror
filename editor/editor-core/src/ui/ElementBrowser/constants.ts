import { gridSize } from '@atlaskit/theme';

export const GRID_SIZE = gridSize();

export const DEVICE_BREAKPOINT_NUMBERS = {
  small: GRID_SIZE * 40,
  medium: GRID_SIZE * 75,
  large: GRID_SIZE * 128,
};

export const SIDEBAR_WIDTH = `${GRID_SIZE * 25}px`;

export const SIDEBAR_HEADING_TEXT = 'Browse';

export const SIDEBAR_HEADING_WRAPPER_HEIGHT = '100px';
export const SIDEBAR_HEADING_PADDING_LEFT = '12px';

export const INLINE_SIDEBAR_HEIGHT = '54px';

export const CATEGORY_ITEM_WIDTH = '218px';

export const SEARCH_ITEM_MARGIN = '12px';
export const SEARCH_ITEM_HEIGHT_WIDTH = '20px';
