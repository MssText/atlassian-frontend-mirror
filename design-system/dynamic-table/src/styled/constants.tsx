import { LegacyRef, ReactNode } from 'react';

import { css } from '@emotion/react';

import { gridSize as getGridSize } from '@atlaskit/theme/constants';

const CSS_VAR_WIDTH = '--local-dynamic-table-width';

const gridSize = getGridSize();

export interface TruncateStyleProps {
  width?: number;
  isFixedSize?: boolean;
  shouldTruncate?: boolean;
  children?: ReactNode;
  testId?: string;
  innerRef?: LegacyRef<HTMLTableCellElement | HTMLTableRowElement> | undefined;
  className?: string;
}

export const truncationWidthStyles = css({ width: `var(${CSS_VAR_WIDTH})` });

export const fixedSizeTruncateStyles = css({
  textOverflow: 'ellipsis',
  whiteSpace: 'nowrap',
});

export const overflowTruncateStyles = css({
  overflow: 'hidden',
});

export const getTruncationStyleVars = ({ width }: TruncateStyleProps) =>
  typeof width !== 'undefined' ? { [CSS_VAR_WIDTH]: `${width}%` } : undefined;

export const cellStyles = css({
  padding: `${gridSize / 2}px ${gridSize}px`,
  border: 'none',
  textAlign: 'left',
  '&:first-of-type': {
    paddingLeft: 0,
  },
  '&:last-child': {
    paddingRight: 0,
  },
});
