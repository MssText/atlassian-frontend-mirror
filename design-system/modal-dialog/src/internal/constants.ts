import { DN30, N30, N800, R400, Y400 } from '@atlaskit/theme/colors';
import { themed } from '@atlaskit/theme/components';
import { gridSize } from '@atlaskit/theme/constants';

export const WIDTH_ENUM: WidthEnumType = {
  values: ['small', 'medium', 'large', 'x-large'],
  widths: {
    small: 400,
    medium: 600,
    large: 800,
    'x-large': 968,
  },
  defaultValue: 'medium',
};

export type WidthNames = 'small' | 'medium' | 'large' | 'x-large';

export interface WidthEnumType {
  values: string[];
  widths: { [index in WidthNames]: number };
  defaultValue: string;
}

export const gutter = 60;

export const verticalOffset = gridSize() * 2;

export const modalPadding = gridSize() * 3;

export const actionMargin = gridSize() / 2;
export const titleMargin = gridSize();

export const keylineHeight = 2;
export const keylineColor = themed({ light: N30, dark: DN30 });

export const focusOutlineColor = N800;
export const dangerColor = R400;
export const warningColor = Y400;
