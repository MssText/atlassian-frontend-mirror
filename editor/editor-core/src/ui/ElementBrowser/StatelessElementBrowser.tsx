import React, { memo } from 'react';
import styled, { css } from 'styled-components';
import { QuickInsertItem } from '@atlaskit/editor-common/provider-factory';
import CategoryItems from './components/CategoryItems';
import CategoryList from './components/CategoryList';
import ElementSearch from './components/ElementSearch';
import {
  DEVICE_BREAKPOINT_NUMBERS,
  GRID_SIZE,
  INLINE_SIDEBAR_HEIGHT,
  SIDEBAR_HEADING_PADDING_LEFT,
  SIDEBAR_HEADING_TEXT,
  SIDEBAR_HEADING_WRAPPER_HEIGHT,
  SIDEBAR_WIDTH,
} from './constants';
import useContainerWidth from './hooks/useContainerWidth';
import { Category, Modes } from './types';

export interface StatelessElementBrowserProps {
  items: QuickInsertItem[];
  categories: Category[];
  onSearch: (searchTerm: string) => void;
  onSelectCategory: (category: Category) => void;
  onSelectItem: (item: QuickInsertItem) => void;
  onEnterKeyPress: (item: QuickInsertItem) => void;
  selectedCategory?: string;
  showSearch: boolean;
  showCategories: boolean;
  mode: keyof typeof Modes;
}

function StatelessElementBrowser(props: StatelessElementBrowserProps) {
  const { containerWidth, ContainerWidthMonitor } = useContainerWidth();
  return (
    <Wrapper>
      <ContainerWidthMonitor />
      {containerWidth < DEVICE_BREAKPOINT_NUMBERS.medium ? (
        <MobileBrowser {...props} />
      ) : (
        <DesktopBrowser {...props} />
      )}
    </Wrapper>
  );
}

const Wrapper = styled.div`
  width: 100%;
  height: 100%;
  max-height: inherit;
  overflow: hidden;
`;

function MobileBrowser({
  showCategories,
  showSearch,
  onSearch,
  mode,
  categories,
  onSelectCategory,
  items,
  onSelectItem,
  onEnterKeyPress,
  selectedCategory,
}: StatelessElementBrowserProps) {
  return (
    <MobileElementBrowserContainer>
      <MobileSideBar showCategories={showCategories}>
        {showSearch && <ElementSearch onSearch={onSearch} mode={mode} />}
        {showCategories && (
          <MobileCategoryListWrapper>
            <CategoryList
              categories={categories}
              onSelectCategory={onSelectCategory}
              selectedCategory={selectedCategory}
            />
          </MobileCategoryListWrapper>
        )}
      </MobileSideBar>
      <MobileMainContent>
        <CategoryItems
          items={items}
          mode={mode}
          onSelectItem={onSelectItem}
          onEnterKeyPress={onEnterKeyPress}
        />
      </MobileMainContent>
    </MobileElementBrowserContainer>
  );
}

function DesktopBrowser({
  showCategories,
  showSearch,
  onSearch,
  mode,
  categories,
  onSelectCategory,
  items,
  onSelectItem,
  onEnterKeyPress,
  selectedCategory,
}: StatelessElementBrowserProps) {
  return (
    <ElementBrowserContainer>
      {showCategories && (
        <SideBar showCategories>
          <SideBarHeading>{SIDEBAR_HEADING_TEXT}</SideBarHeading>
          <CategoryListWrapper>
            <CategoryList
              categories={categories}
              onSelectCategory={onSelectCategory}
              selectedCategory={selectedCategory}
            />
          </CategoryListWrapper>
        </SideBar>
      )}
      <MainContent>
        {showSearch && (
          <SearchContainer>
            <ElementSearch onSearch={onSearch} mode={mode} />
          </SearchContainer>
        )}
        <CategoryItems
          items={items}
          mode={mode}
          onSelectItem={onSelectItem}
          onEnterKeyPress={onEnterKeyPress}
        />
      </MainContent>
    </ElementBrowserContainer>
  );
}

const baseBrowserContainerStyles = css`
  display: flex;
  height: 100%;
`;

const MobileElementBrowserContainer = styled.div`
  ${baseBrowserContainerStyles};
  max-height: inherit;
  flex-direction: column;
`;

const ElementBrowserContainer = styled.div`
  ${baseBrowserContainerStyles};
  /**
   * For Safari, as 100% doesn't work.
   */
  max-height: fill-available;
  flex-direction: row;
`;

type SideBarType = {
  showCategories: boolean;
};

const baseSidebarStyles = css`
  display: flex;
  flex-direction: column;

  overflow-x: auto;
  overflow-y: hidden;
`;

const MobileSideBar = styled.div`
  ${baseSidebarStyles};
  padding: ${GRID_SIZE}px;
  padding-top: 0;
  flex: 0 0
    ${({ showCategories }: SideBarType) =>
      showCategories ? 'auto' : INLINE_SIDEBAR_HEIGHT};
`;

const SideBar = styled.div`
  ${baseSidebarStyles};
  flex: 0 0
    ${({ showCategories }: SideBarType) =>
      showCategories ? SIDEBAR_WIDTH : 'auto'};
`;

const SideBarHeading = styled.h2`
  flex: 0 0 ${SIDEBAR_HEADING_WRAPPER_HEIGHT};
  display: inline-flex;
  align-items: center;
  padding-left: ${SIDEBAR_HEADING_PADDING_LEFT};
  font-weight: 700;
`;

/**
 *  In enzyme styled components show up as styled.element
 *  and if we don't wanna export SideBarHeading just for testing.
 *  https://github.com/styled-components/styled-components/issues/896
 */
SideBarHeading.displayName = 'SideBarHeading';

const MobileMainContent = styled.div`
  flex: 1 1 auto;

  display: flex;
  flex-direction: column;

  overflow-y: auto;

  /** Needed for Safari to work with current css.
  * For some reason, specified 100% height in parent containers didn't work and
  * the content still overflows in safari.
  */
  max-height: -webkit-fill-available;
`;

const MainContent = styled(MobileMainContent)`
  margin-left: ${GRID_SIZE * 2}px;
`;

const SearchContainer = styled.div`
  padding-bottom: ${GRID_SIZE * 3}px;
`;

const MobileCategoryListWrapper = styled.nav`
  display: flex;
  overflow-x: auto;

  padding: ${GRID_SIZE}px 0;

  overflow: -moz-scrollbars-none;
  ::-webkit-scrollbar {
    display: none;
  }
  scrollbar-width: none;
  -ms-overflow-style: none;
`;

const CategoryListWrapper = styled(MobileCategoryListWrapper)`
  padding: 0;
  margin-top: ${GRID_SIZE * 4}px;
  height: 100%;
  flex-direction: column;
`;

const MemoizedElementBrowser = memo(StatelessElementBrowser);

export default MemoizedElementBrowser;
