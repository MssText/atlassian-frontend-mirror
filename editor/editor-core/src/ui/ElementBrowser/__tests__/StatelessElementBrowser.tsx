const getWidth = jest.fn();

jest.mock('../hooks/useContainerWidth', () => () => ({
  containerWidth: getWidth(),
  ContainerWidthMonitor: jest.fn(),
}));
import React from 'react';
import { shallow } from 'enzyme';
import { DEVICE_BREAKPOINT_NUMBERS } from '../constants';
import StatelessElementBrowser from '../StatelessElementBrowser';

export const testProps = {
  items: [{ name: 'item-1', title: 'Item 1', action: jest.fn() }],
  categories: [{ name: 'category-1', title: 'Category 1' }],
  selectedCategory: 'all',
  onSearch: jest.fn,
  onSelectCategory: jest.fn,
  onSelectItem: jest.fn,
  onEnterKeyPress: jest.fn,
  showSearch: true,
  showCategories: true,
};

describe('StatelessElementBrowser', () => {
  it('should render mobile components for width < 600', () => {
    getWidth.mockReturnValueOnce(DEVICE_BREAKPOINT_NUMBERS.small);
    const wrapper = shallow(
      <StatelessElementBrowser {...testProps} mode="inline" />,
    );
    expect(wrapper.find('MobileBrowser').shallow()).toMatchSnapshot();
  });
  it('should should not render desktop components on mobile', () => {
    getWidth.mockReturnValueOnce(DEVICE_BREAKPOINT_NUMBERS.small);
    const wrapper = shallow(
      <StatelessElementBrowser {...testProps} mode="inline" />,
    );
    expect(wrapper.find('DesktopBrowser')).toHaveLength(0);
  });
  it('should render desktop components for width >= 600', () => {
    getWidth.mockReturnValueOnce(DEVICE_BREAKPOINT_NUMBERS.medium);
    const wrapper = shallow(
      <StatelessElementBrowser {...testProps} mode="full" />,
    );
    expect(wrapper.find('DesktopBrowser').shallow()).toMatchSnapshot();
  });
  it('should should not render mobile components on desktop', () => {
    getWidth.mockReturnValueOnce(DEVICE_BREAKPOINT_NUMBERS.medium);
    const wrapper = shallow(
      <StatelessElementBrowser {...testProps} mode="full" />,
    );
    expect(wrapper.find('MobileBrowser')).toHaveLength(0);
  });
  it('should not render a sidebar heading on mobile', () => {
    getWidth.mockReturnValueOnce(DEVICE_BREAKPOINT_NUMBERS.small);
    const wrapper = shallow(
      <StatelessElementBrowser {...testProps} mode="inline" />,
    );
    expect(
      wrapper
        .find('MobileBrowser')
        .shallow()
        .find('SideBarHeading'),
    ).toHaveLength(0);
  });
  it('should render a sidebar heading on desktop', () => {
    getWidth.mockReturnValueOnce(DEVICE_BREAKPOINT_NUMBERS.medium);
    const wrapper = shallow(
      <StatelessElementBrowser {...testProps} mode="full" />,
    );
    expect(
      wrapper
        .find('DesktopBrowser')
        .shallow()
        .find('SideBarHeading'),
    ).toHaveLength(1);
  });
});
