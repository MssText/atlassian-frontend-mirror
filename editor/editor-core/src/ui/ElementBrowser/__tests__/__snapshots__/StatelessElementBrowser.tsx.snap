// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatelessElementBrowser should render desktop components for width >= 600 1`] = `
<styled.div>
  <styled.div
    showCategories={true}
  >
    <SideBarHeading>
      Browse
    </SideBarHeading>
    <Styled(styled.nav)>
      <Memo(WithAnalyticsContext(CategoryList))
        categories={
          Array [
            Object {
              "name": "category-1",
              "title": "Category 1",
            },
          ]
        }
        onSelectCategory={[Function]}
        selectedCategory="all"
      />
    </Styled(styled.nav)>
  </styled.div>
  <Styled(styled.div)>
    <styled.div>
      <Memo(WithAnalyticsContext(ElementSearch))
        mode="full"
        onSearch={[Function]}
      />
    </styled.div>
    <Memo(WithAnalyticsContext(CategoryItems))
      items={
        Array [
          Object {
            "action": [MockFunction],
            "name": "item-1",
            "title": "Item 1",
          },
        ]
      }
      mode="full"
      onEnterKeyPress={[Function]}
      onSelectItem={[Function]}
    />
  </Styled(styled.div)>
</styled.div>
`;

exports[`StatelessElementBrowser should render mobile components for width < 600 1`] = `
<styled.div>
  <styled.div
    showCategories={true}
  >
    <Memo(WithAnalyticsContext(ElementSearch))
      mode="inline"
      onSearch={[Function]}
    />
    <styled.nav>
      <Memo(WithAnalyticsContext(CategoryList))
        categories={
          Array [
            Object {
              "name": "category-1",
              "title": "Category 1",
            },
          ]
        }
        onSelectCategory={[Function]}
        selectedCategory="all"
      />
    </styled.nav>
  </styled.div>
  <styled.div>
    <Memo(WithAnalyticsContext(CategoryItems))
      items={
        Array [
          Object {
            "action": [MockFunction],
            "name": "item-1",
            "title": "Item 1",
          },
        ]
      }
      mode="inline"
      onEnterKeyPress={[Function]}
      onSelectItem={[Function]}
    />
  </styled.div>
</styled.div>
`;
